arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1712 @@
+{
+	"swagger": "2.0",
+	"info": {
+		"version": "2016-02-03",
+		"title": "iotHubClient",
+		"description": "API for using the IoT Hub Features"
+	},
+	"host": "management.azure.com",
+	"schemes": ["https"],
+	"paths": {
+		"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}": {
+			"get": {
+				"tags": ["GET"],
+				"summary": "Get an IotHub",
+				"description": "Get an IotHub",
+				"operationId": "IotHubResource_GetIotHub",
+				"consumes": [],
+				"produces": ["application/json", "text/json"],
+				"parameters": [{
+					"name": "resourceName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "subscriptionId",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/IotHubDescription"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			},
+			"put": {
+				"tags": ["PUT"],
+				"operationId": "IotHubResource_CreateOrUpdateIotHub",
+				"consumes": ["application/json", "text/json"],
+				"produces": ["application/json", "text/json"],
+				"parameters": [{
+					"name": "subscriptionId",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "iotHubDescription",
+					"in": "body",
+					"required": true,
+					"schema": {
+						"$ref": "#/definitions/IotHubDescription"
+					}
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/IotHubDescription"
+						}
+					},
+					"201": {
+						"description": "Created",
+						"schema": {
+							"$ref": "#/definitions/IotHubDescription"
+						}
```

Comment:
By returning 201 and 200 on a create, this fits into the pattern of long running operation. If this is indeed one, then please apply this extension https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-long-running-operation